apps/src/code-studio/pd/application/principalApproval1819/PrincipalApproval1819Application.jsx

```diff
@@ -0,0 +1,235 @@
+import ApplicationFormComponent from '../ApplicationFormComponent';
+import SchoolAutocompleteDropdown from '@cdo/apps/templates/SchoolAutocompleteDropdown';
+import {isInt, isPercent} from '@cdo/apps/util/formatValidation';
+
+const RACE_LIST = ['white', 'black', 'hispanic', 'asian', 'pacificIslander', 'americanIndian', 'other'];
+
+export default class PrincipalApproval1819Application extends FormController {
+  static propTypes = {
+    ...FormController.propTypes,
+    teacherApplication: PropTypes.shape({
+      course: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      application_guid: PropTypes.string.isRequired
+    }).isRequired
+  }
+  /**
+   * @override
+   */
+  getPageComponents() {
+    return [
+      PrincipalApprovalComponent
+    ];
+  }
+
+  /**
+   * @override
+   */
+  getPageProps() {
+    return {
+      ...super.getPageProps(),
+      teacherApplication: this.props.teacherApplication
+    };
+  }
+
+  /**
+   * @override
+   */
+  serializeFormData() {
+    return {
+      ...super.serializeFormData(),
+      application_guid: this.props.teacherApplication.application_guid
+    };
+  }
+
+  onSuccessfulSubmit() {
+    window.location.reload(true);
+  }
+}
+
+class PrincipalApprovalComponent extends ApplicationFormComponent {
+  static labels = PageLabels;
+
+  static associatedFields = [
+    ...Object.keys(PageLabels)
+  ];
+
+  handleSchoolChange = selectedSchool => {
+    this.handleChange({school: selectedSchool && selectedSchool.value});
+  };
+
+  renderSchoolSection() {
+    // TODO: Mehal - this should be a seperate component
+    return (
+      <div>
+        <FormGroup
+          id="school"
+          controlId="school"
+          validationState={this.getValidationState("school")}
+        >
+          <Row>
+            <Col md={6}>
+              <ControlLabel>
+                School
+                <span style={{color: 'red'}}> *</span>
+              </ControlLabel>
+            </Col>
+          </Row>
+          <Row>
+            <Col md={6}>
+              <SchoolAutocompleteDropdown
+                value={this.props.data.school}
+                onChange={this.handleSchoolChange}
+              />
+            </Col>
+          </Row>
+        </FormGroup>
+        {
+          this.props.data.school && this.props.data.school === '-1' &&
+          <div>
+            {this.inputFor("schoolName")}
+            {this.inputFor("schoolAddress")}
+            {this.inputFor("schoolCity")}
+            {this.selectFor("schoolState", {placeholder: "Select a state"})}
+            {this.inputFor("schoolZipCode")}
+            {this.radioButtonsFor("schoolType")}
+          </div>
+        }
+      </div>
+    );
+  }
+
+  renderCourseReplacementSection() {
+    if (this.props.teacherApplication.course === 'Computer Science Discoveries') {
+      return this.checkBoxesWithAdditionalTextFieldsFor('replaceWhichCourseCsp', {
```

Comment:
Shouldn't this one be `Csd`?

Reply:
Presuming these are different fields depending on which course is selected, we should also add `getDynamicallyRequiredFields` and `processPageData` method overrides to require the included field and clear leftover answers on the excluded one.